src/com/amplitude/api/Utils.java

```diff
@@ -0,0 +1,47 @@
+package com.amplitude.api;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/**
+ * Created by danieljih on 4/18/16.
+ */
+public class Utils {
+
+    public static final String TAG = "com.amplitude.api.Utils";
+    private static AmplitudeLog logger = AmplitudeLog.getLogger();
+
+
+    /**
+     * Do a shallow copy of a JSONObject. Takes a bit of code to avoid
+     * stringify and reparse given the API.
+     */
+    static JSONObject cloneJSONObject(final JSONObject obj) {
+        if (obj == null) {
+            return null;
+        }
+
+        // obj.names returns null if the json obj is empty.
+        JSONArray nameArray = null;
+        try {
+            nameArray = obj.names();
+        } catch (ArrayIndexOutOfBoundsException e) {
```

Comment:
No idea, but we've been using it for a while now. I think the other way to clone JSONObjects is to do like a new JSONObject(old.toString()) or something, which is why this method was added